src/main/java/mekanism/common/tile/base/TileEntityMekanism.java

```diff
@@ -431,8 +438,38 @@ public void addComponent(ITileComponent component) {
     }
 
     @Nonnull
+    @Override
+    @SuppressWarnings("ConstantConditions")
     public Component getName() {
-        return TextComponentUtil.translate(Util.makeDescriptionId("container", getBlockType().getRegistryName()));
+        return hasCustomName() ? getCustomName() : TextComponentUtil.build(getBlockType());
+    }
+
+    @Nonnull
+    @Override
+    @SuppressWarnings("ConstantConditions")
+    public Component getDisplayName() {
+        return hasCustomName() ? getCustomName() : TextComponentUtil.translate(Util.makeDescriptionId("container", getBlockType().getRegistryName()));
```

Comment:
I added this as a different implementation than getName just for more "accurate" results at least for purposes as we use display name for the container, but I am wondering if this should maybe instead be:
```java
if (isNameable()) {
//current return
}
return TextComponentUtil.build(getBlockType());
```
Thoughts? Alternatively maybe move the logic of this return to AttributeGUI and remove the getDisplayName override allowing it to be the default in case mods make assumptions and want the actual name not the name for displaying in containers.

Reply:
I think that we should wrap that return inside the `isNameable`, because if it isn't showing the custom name is useless.